### PR TITLE
Spark 3.3: Add doc for the changelog view procedure.

### PR DESCRIPTION
**Pull request by flyrain**
cc @aokolnychyi @RussellSpitzer @chenjunjiedada @rdblue

### CODE REVIEW

File: docs/spark-procedures.md [hunk 1]
```diff
@@ -587,3 +587,75 @@ Get all the snapshot ancestors by a particular snapshot
...
+| `options`     |   | map<string, string> | A map of Spark read options to use. For example, `start-snapshot-id`, the snapshot id to start reading from exclusively. If not provided, the table’s first snapshot will be used as the starting point. `end-snapshot-id`, the snapshot id to stop reading at inclusively. Default to the current snapshot. `start-timestamp`, the timestamp to start reading from exclusively. If not provided, the table’s first snapshot will be used as the starting point.`end-timestamp`, the timestamp to stop reading at inclusively. If not provided, the table’s current snapshot will be used as the ending point. |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:603]: I think this ends up being hard to read, maybe we should have a small "options" table below here

File: docs/spark-procedures.md [hunk 2]
```diff
@@ -587,3 +587,75 @@ Get all the snapshot ancestors by a particular snapshot
...
+|`compute_updates`| | boolean | Whether to compute updates. Defaults to false                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                   |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:604]: This is a bit confusing. I think looking at this from an outside perspective i'm not sure what "compute_updates" would mean. Ideally all of our descriptions should be a more descriptive than a restatement of the name of the column.

File: docs/spark-procedures.md [hunk 3]
```diff
@@ -587,3 +587,75 @@ Get all the snapshot ancestors by a particular snapshot
...
+| `table`       | ✔️ | string | Name of the table to create changlog view                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                       |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:601]: The source table for the change log?

File: docs/spark-procedures.md [hunk 4]
```diff
@@ -587,3 +587,75 @@ Get all the snapshot ancestors by a particular snapshot
...
+|`identifier_columns`| | array<string> | The list of identifier columns. If not provided, and `compute_updates` is true, the table’s current identifier fields will be used.                                                                                                                                                                                                                                                                                                                                                                                                                                                             |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:605]: Description should contain details on how these columns are used. In this case, 
"Used when compute_updates is true to identify pre and post update rows. If a delete and insert record have identical identifier columns, they are considered to be pre and post updates of the same row."

Or something like that

File: docs/spark-procedures.md [hunk 5]
```diff
@@ -587,3 +587,75 @@ Get all the snapshot ancestors by a particular snapshot
...
+|`remove_carryovers`| | boolean | Whether to remove carry-over rows. Defaults to true.                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                                            |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:606]: carry_over rows is not defined and similar comment to above options. Each description need to explain the usage and function

File: docs/spark-procedures.md [hunk 6]
```diff
@@ -587,3 +587,75 @@ Get all the snapshot ancestors by a particular snapshot
...
+
+#### Output
+| Output Name | Type | Description |
+| ------------|------|-------------|
+| `changelog_view` | string | The name of the changelog view |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:611]: name of the created changelog view

**Review by RussellSpitzer — COMMENTED** [review]
Left some comments on the arg descriptions. I think we need to clean this up a bit so that users can really take advantage of this great feature

**Comment by flyrain** [comment]
Hi @RussellSpitzer, resolved comments. It's ready for another look.

File: docs/spark-procedures.md [hunk 7]
```diff
@@ -587,3 +587,119 @@ Get all the snapshot ancestors by a particular snapshot
...
+
+In this case, the procedure marks the row before the update as an `UPDATE_BEFORE` image and the row after the update
+as an `UPDATE_AFTER` image, resulting in the following pre/post update images:
+
+| id  | name   | _change_type |
+|-----|--------|--------------|
+| 3   | Robert | UPDATE_BEFORE|
+| 3   | Dan    | UPDATE_AFTER |
+
+#### Usage
```
Comment (CodingCat) [inline docs/spark-procedures.md:635]: I think we may want to move the basic usage as the first subsection under `create_changelog_view` , and then some special subsections to explain what's a carry-over row and what is a `pre/post update` and (how to configure it with some examples)....which is a more straightforward tutorial structure?

File: docs/spark-procedures.md [hunk 8]
```diff
@@ -587,3 +587,119 @@ Get all the snapshot ancestors by a particular snapshot
...
+| `table`       | ✔️ | string | Name of the source table for the changelog                                                                                                                                                                            |
+| `changelog_view`        |   | string | Name of the view to create                                                                                                                                                                                            |
+| `options`     |   | map<string, string> | A map of Spark read options to use                                                                                                                                                                                    |
+|`compute_updates`| | boolean | Whether to compute pre/post update images, defaults to false.                                                                                                                                                         |
```
Comment (RussellSpitzer) [inline docs/spark-procedures.md:642]: update images (see below for more information)

**Review by RussellSpitzer — APPROVED** [review]
I think this is a good overview but I agree with @CodingCat to reorder the document so that Usage, Output and Examples comes first. Then "Carry-Over Rows" and "Pre/Post Update Images"

I think those section titles also don't need verbs but that's a nit.

**Comment by flyrain** [comment]
Thanks @RussellSpitzer @CodingCat for the reviews!

**Pull request merged**